landoapi/transplants.py

```diff
@@ -301,6 +303,33 @@ def warning_wip_commit_message(*, revision, **kwargs):
         return "This revision is marked as a WIP. Please remove `WIP:` before landing."
 
 
+@RevisionWarningCheck(8, "Repository is under a soft code freeze.", True)
+def warning_code_freeze(*, repo, **kwargs):
+    repo_fields = repo.get("fields", {})
+    short_name = repo_fields.get("shortName")
+    if short_name == "mozilla-central":
+        product_details = requests.get(
+            "https://product-details.mozilla.org/1.0/firefox_versions.json"
+        ).json()
```

Comment:
Good to wrap this with a try/except in case this service is down. Caught error can alter the warning message to say that soft freeze state could not be checked, for example.